pallets/loans/src/lib.rs

```diff
@@ -519,16 +519,45 @@ pub mod pallet {
 			interest_rate_per_sec: T::Rate,
 			loan_type: LoanType<T::Rate, T::Balance>,
 		) -> DispatchResultWithPostInfo {
-			// ensure sender has the pricing admin role in the pool
-			ensure_role!(pool_id, origin, PoolRole::PricingAdmin);
+			let owner = ensure_signed(origin)?;
+
 			let active_count =
-				Self::price_loan(pool_id, loan_id, interest_rate_per_sec, loan_type)?;
+				Loan::<T>::try_mutate(pool_id, loan_id, |loan| -> Result<u32, DispatchError> {
+					let loan = loan.as_mut().ok_or(Error::<T>::MissingLoan)?;
+
+					match loan.status {
+						LoanStatus::Created => {
+							Self::ensure_role(pool_id, owner, PoolRole::PricingAdmin)?;
+							let active_count = Self::price_created_loan(
+								pool_id,
+								loan_id,
+								interest_rate_per_sec,
+								loan_type,
+							);
+
+							loan.status = LoanStatus::Active;
+							active_count
+						}
+						LoanStatus::Active => {
+							Self::ensure_role(pool_id, owner, PoolRole::LoanAdmin)?;
+							Self::price_active_loan(
+								pool_id,
+								loan_id,
+								interest_rate_per_sec,
+								loan_type,
+							)
+						}
+						LoanStatus::Closed { .. } => Err(Error::<T>::LoanIsActive)?,
```

Comment:
Previously only created loans could be priced, now created and active loans. Hence, this error should be renamed to something like
```suggestion
						LoanStatus::Closed { .. } => Err(Error::<T>::LoanIsClosed)?,
```

Reply:
It makes total sense!